drv/stm32xx-i2c-server/src/main.rs

```diff
@@ -452,6 +513,14 @@ fn configure_pins(
     let sys = SYS.get_task_id();
     let sys = Sys::from(sys);
 
+    //
+    // Before we conbfigure our pins, wiggle SCL to shake off any old
+    // transaction.
+    //
+    for ndx in (0..pins.len()).step_by(2) {
```

Comment:
This seems a little fragile – I think it would be worth revamping the existing `I2CPin` data type to contain SDA and SCL by name, e.g.

```rust
pub struct I2cPins {
    pub controller: drv_i2c_api::Controller,
    pub port: drv_i2c_api::PortIndex,
    pub sda: sys_api::PinSet,
    pub scl: sys_api::PinSet,
    pub function: sys_api::Alternate,
}
```

(also renamed to `I2cPins` to be more obvious)

Reply:
Yes, agreed! Code is definitely cleaner; thank you!